pkgs/servers/x11/xorg/default.nix

```diff
@@ -71,7 +76,8 @@ lib.makeScope newScope (self: with self; {
   }) {};
 
   fontadobe100dpi = callPackage ({ stdenv, pkg-config, fetchurl, bdftopcf, fontutil, mkfontscale }: stdenv.mkDerivation {
-    name = "font-adobe-100dpi-1.0.3";
+    pname = "font-adobe";
+    version = "100dpi-1.0.3";
```

Comment:
@grahamc please check if this is wanted.

Reply:
With all due respect, you say there are *now* multiple `font-adobe` packages, but that's already the case per the [package index](https://search.nixos.org/packages?channel=21.05&from=0&size=50&sort=relevance&query=font-adobe) precisely because the new regex mirrors what Nix already does.

Now in my opinion, this variant has the advantage of maintaining the status quo in the package index which could be desirable. However, the good thing about pname+version is that it's more flexible: we can decide case-by-case what's part of the version number and what isn't, rather than relying on a one-size-fits-all regex. (I would also argue this regex is kind of an heuristic to begin with) Now if I were to do it from scratch, I would look at [upstream](https://gitlab.freedesktop.org/xorg/font/adobe-100dpi) and see what they consider to be the version number. Each variant comes from a different repository, and in [multiple commits](https://gitlab.freedesktop.org/xorg/font/adobe-100dpi/-/commit/db9920040c4f3a8479137515c8730c85fc724fee), there is a space before the version number. In the configure.ac, it also suggests that the version number is 1.0.3, so I consider that to be canonical.

Thus I would disagree slightly with @grahamc and suggest that, for this particular set of package, the initial regex fits better than the default one. It is also simpler, thus it should be used. Assuming of course that it doesn't break anything, but I think that's unlikely since it's metadata.

Reply:
What about using the string until the last minus as the package name.
The version number after the first minus with a digit, as proposed by @grahamc.
I think this would not change the current behavior. But it would prevent issues with `-rc1`